yarn-project/simulator/src/public/public_execution_context.ts

```diff
@@ -134,6 +134,7 @@ export class PublicExecutionContext extends TypedOracle {
     // TODO(https://github.com/AztecProtocol/aztec-packages/issues/885)
     this.unencryptedLogs.push(log);
     this.log.verbose(`Emitted unencrypted log: "${log.toHumanReadable()}"`);
+    return Fr.fromBuffer(log.hash());
```

Comment:
Is this oracle actually used from private/public context?

Reply:
Yes, in the public context, when public fns emit encrypted logs (e.g. the `child` contract used in ordering test).

Reply:
Ah I see, the calls are inverted (emit_unencrypted_log calls the oracle and the context). I think ultimately we'll need it to be the other way around: contracts call context.emit_unencrypted_log and that one calls the right context). That way, you'll be able to make it work with both PublicContext and AvmContext (behind the PublicContextInterface abstraction).

I can make the inversion myself so that your PR is not blocked, just make sure that it will be possible. I think you'll probably want to delete accumulate_unencrypted_logs if it's not used by anyone.

Reply:
Thanks - I'm happy to do the changes but am not that familiar with `AvmContext` at all. Either way, I will remove `accumulate_unencrypted_logs` as it's not used in `PublicContext`. For clarity then it's probably best I rename `AvmContext.accumulate_unencrypted_logs` to `AvmContext.push...` or `AvmContext.emit...`, any preference?

Reply:
👍 no preferences for the renaming!